From 96b1ae8ac47a2c9ce171f700da8957152613bf02 Mon Sep 17 00:00:00 2001
From: Eric Naim <dnaim@cachyos.org>
Date: Wed, 25 Mar 2026 17:46:59 +0800
Subject: [PATCH 041/243] CACHY: mm: lru-gen: Protect the working set of the
 last 100 jiffies

This enables MGLRU's thrashing prevention and prevents freezes when
system memory is under high pressure.

Signed-off-by: Eric Naim <dnaim@cachyos.org>
---
 mm/vmscan.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/mm/vmscan.c b/mm/vmscan.c
index 6f828844b0e8..a7df897d61c3 100644
--- a/mm/vmscan.c
+++ b/mm/vmscan.c
@@ -4133,7 +4133,11 @@ static bool lruvec_is_reclaimable(struct lruvec *lruvec, struct scan_control *sc
 }
 
 /* to protect the working set of the last N jiffies */
+#ifdef CONFIG_CACHY
+static unsigned long lru_gen_min_ttl __read_mostly = 100;
+#else
 static unsigned long lru_gen_min_ttl __read_mostly;
+#endif
 
 static void lru_gen_age_node(struct pglist_data *pgdat, struct scan_control *sc)
 {
-- 
2.55.0

